### ISSUE TITLE
0 Supported Devices on Google play when using this SDK

### ISSUE THREAD

**Issue by davin12x** [nestlabs/android-sdk#28]
After changing to
compile 'commons-io:commons-io:2.4'
 I am still having this issue. If I remove Nest SDK everything works fine

compile 'com.android.support:appcompat-v7:25.3.1'
    compile 'com.android.support:design:25.3.1'
    compile 'com.android.support:support-v4:25.3.1'
    compile 'commons-io:commons-io:2.4'
    compile 'com.nestlabs:android-sdk:1.1.1'

**Comment by harenj** [comment]
Issue #18 is the same thing, I resolved it there. This problem shouldn't exist anymore.

**Comment by davin12x** [comment]
Yes . But the problem still exists . as soon as i add nest sdk it imports 
 org.apache.commons.io.2.4 and commons-io-2.4. 
it is suppose to import only commons -io-2.4 not  org.apache.commons.io.2.4.

**Comment by harenj** [comment]
> as soon as i add nest sdk it imports org.apache.commons.io.2.4 and commons-io-2.4.

What do you mean by this? Can you paste your project's build.gradle file? Also, what release of the Nest SDK are you compiling against? The latest version is `1.1.0`. See https://github.com/nestlabs/android-sdk/releases/tag/1.1.0

**Comment by davin12x** [comment]
yes , I tried with 1.1.1 and 1.1.0 too. 
![gradle](https://cloud.githubusercontent.com/assets/6825007/25665284/231a3796-2feb-11e7-94fe-8c289afc5f61.png)

and this is what i got after uploading 
![uplaod error](https://cloud.githubusercontent.com/assets/6825007/25665393/6a0ec20c-2feb-11e7-82f5-c86bf94122ca.png)

As i said, as soon as i remove nest sdk dependency . It works then.

**Comment by harenj** [comment]
You shouldn't have to put `compile 'commons-io:commons-io:2.4' in your `build.gradle` file. It is inside the Nest SDK `build.gradle` and you will pick it up. Can you try removing it from your project's `build.gradle` and see if the issue resolves itself?

**Comment by davin12x** [comment]
The issue is still there after removing `compile 'commons-io:commons-io:2.4'` from my project build.gradle file and reason is https://github.com/nestlabs/android-sdk/releases/tag/1.1.0 Nest SDK source code has `compile 'org.apache.directory.studio:org.apache.commons.io:2.4'` instead of `compile 'commons-io:commons-io:2.4'`

**Comment by harenj** [comment]
> Nest SDK source code has `compile 'org.apache.directory.studio:org.apache.commons.io:2.4'`

I see `compile 'commons-io:commons-io:2.4` in the Nest `build.gradle` here: https://github.com/nestlabs/android-sdk/blob/master/sdk/build.gradle#L203

**Comment by davin12x** [comment]
`https://github.com/nestlabs/android-sdk/blob/master/sdk/build.gradle#L203` refers to the latest master not the released version of Nest SDK

Below are released versions 
`https://github.com/nestlabs/android-sdk/blob/1.1.1/sdk/build.gradle`
`https://github.com/nestlabs/android-sdk/blob/1.1.0/sdk/build.gradle`
and both of them are having 
`compile 'org.apache.directory.studio:org.apache.commons.io:2.4'`

Issue is fixed on 
![fixed-date](https://cloud.githubusercontent.com/assets/6825007/25783519/412d5e96-332b-11e7-824e-3f414ed4872d.png)

But release date is on 
![release date](https://cloud.githubusercontent.com/assets/6825007/25783525/50568f0a-332b-11e7-8adc-c83b5351a74c.png)

So what I m saying is fix is not available in any of the released version.

**Comment by harenj** [comment]
Ah, I understand.

You can build against the mainline version of the Nest SDK using [Jitpack](https://jitpack.io/). This will allow you to specify a version of the tree that has not yet been associated with a release tag. Alternatively, feel free to use my fork which has the fix. However, I cannot guarantee the stability and maintenance of this fork going forward.

`compile 'com.github.harenj:android-sdk:mainline-SNAPSHOT'`

**Comment by davin12x** [comment]
Ok. I also see that if i build using Jitpack against mainline it has "minifyEnabled true" which i need to disable first to make the nest sdk working properly. Anyways thanks for help.

**Comment by harenj** [comment]
@davin12x it looks like the latest commit has the fix. Can you check? https://github.com/nestlabs/android-sdk/blob/master/sdk/build.gradle#L185
